sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -816,6 +817,9 @@ def list_blobs(self, name_starts_with=None, include=None, **kwargs):
         if include and not isinstance(include, list):
             include = [include]
 
+        if kwargs.pop('prefix', None):
+            warnings.warn("Passing 'prefix' has no effect on filtering, " +
+                          "please use the 'name_starts_with' parameter instead.")
```

Comment:
Wondering if we might be better off just adopting `prefix` as a supported parameter?
It's difficult to tell how much this "API rename" has been a hinderance, because it's those who are familiar with the REST API that trip up on it, whereas it's impossible to tell home many others have discovered/used this successfully...

Reply:
Hi @annatisch, I would not be opposed to this. My only question is whether this would be considered breaking or not? Although `prefix` does nothing as it currently stands, technically someone who _was_ using `prefix` will now actually have filtering suddenly introduced. So curious to hear your thoughts there.

As a follow-up, if we do just accept `prefix`, do we want to add any warnings or turn people away from `name_starts_with`, as what if a customer provides both `prefix` and `name_starts_with`, what should take precedence?